pkg/flow/internal/controller/loader.go

```diff
@@ -117,7 +127,7 @@ func NewLoader(opts LoaderOptions) *Loader {
 // The provided parentContext can be used to provide global variables and
 // functions to components. A child context will be constructed from the parent
 // to expose values of other components.
```

Comment:
For code readability, the comments on `Apply` method should mention what is going on with `configBlocks`, `declares` and what is `parentModuleDefinitions` for.